### ISSUE TITLE
one or more objects failed to apply, reason: HPECSIDriver.storage.hpe.com "csi-driver" is invalid: spec.disableNodeConfiguration: Required value (retried 2 times).

### ISSUE THREAD

**Issue by nin0-0** [hpe-storage/csi-driver#364]
Hi, 

I'm trying to deploy the CRD with argocd but I have a weird situation. When I sync the csi-driver [HPECSIDriver](https://scod.hpedev.io/partners/redhat_openshift/index.html#openshift_cli) I get the error

`one or more objects failed to apply, reason: HPECSIDriver.storage.hpe.com "csi-driver" is invalid: spec.disableNodeConfiguration: Required value (retried 2 times).`

When I add it, the sync works but in the logs it says disableNodeConfiguration is not in the schema:

```
oc logs -n hpe-csi-driver pods/hpe-csi-driver-operator-75df9f6664-sjvb8
{
  "level": "error",
  "ts": 1700841795.6133084,
  "msg": "Reconciler error",
  "controller": "hpecsidriver-controller",
  "object": {
    "name": "csi-driver",
    "namespace": "hpe-csi-driver"
  },
  "namespace": "hpe-csi-driver",
  "name": "csi-driver",
  "reconcileID": "7b56cc4c-1d1b-4626-8ec7-fd1299f5f140",
  "error": "failed to install release: values don't meet the specifications of the schema(s) in the following chart(s):\nhpe-csi-driver:\n- (root): Additional property disableNodeConfiguration is not allowed\n",
  "stacktrace": "sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:326\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:234"
}
```

Am I doing something wrong?

**Comment by grant-cl** [comment]
have the same issue, fresh install of openshift 4.12.42 pulled in hpe 2.4.0 after adding disableNodeConfiguration to the HPECSIDriver we are stuck on: Additional property disableNodeConfiguration is not allowed

**Comment by datamattsson** [comment]
We pushed the wrong Operator bundle. Red Hat's CI didn't catch it. Red Hat is trying to fix it. Hopefully this week. Install the Helm chart if you're on 4.13 or 4.14 in the meantime. 4.12 should still run HPE CSI Driver 2.3.0.

**Comment by nin0-0** [comment]
`hpe-csi-operator.v2.4.0` is available for ocp 4.12 too

**Comment by nin0-0** [comment]
I've been trying to install v.2.3.0 but somehow it's only able to install 2.4.0 - and it's not super clear how the version should be annotated (v2.3.0 or 2.3.0).

In the 2.4.0 description of the subscription I see:
`    Message:               constraints not satisfiable: no operators found in package hpe-csi-operator in the catalog referenced by subscription hpe-csi-operator, subscription hpe-csi-operator exists`

So okay, I guessed maybe 2.4.0 is not available, so I patched the subs, but same result. Also cleanly deleted all subs, crds, but same result.

Any update on this? It's related to the operator publishing mistake I'm guessing?
